### PR TITLE
Modern installation full Doctrine

### PR DESCRIPTION
**Pull request by jolelievre**
Only uses doctrine to install and remove the database tables
Add missing entities to ensure the database is correctly initialized

These new features require PS 1.7.7

### CODE REVIEW

File: src/Entity/ProductCommentCriterion.php [hunk 1]
```diff
@@ -108,4 +121,25 @@ public function setActive($active)
 
         return $this;
     }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getLanguages(): ArrayCollection
```
Comment (PierreRambaud) [inline src/Entity/ProductCommentCriterion.php]: Watchout, can be also a PersistentCollection (happened for me in a side project :sweat_smile: )

File: productcomments.php [hunk 2]
```diff
@@ -57,25 +58,15 @@ public function __construct()
         $this->displayName = $this->trans('Product Comments', [], 'Modules.Productcomments.Admin');
         $this->description = $this->trans('Allows users to post reviews and rate products on specific criteria.', [], 'Modules.Productcomments.Admin');
 
-        $this->ps_versions_compliancy = array('min' => '1.7.6', 'max' => _PS_VERSION_);
+        $this->ps_versions_compliancy = array('min' => '1.7.7', 'max' => _PS_VERSION_);
```
Comment (Progi1984) [inline productcomments.php:61]: Please change the major version if you change the compliancy.

File: src/Database/DatabaseUpdater.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
```
Comment (Progi1984) [inline src/Database/DatabaseUpdater.php:26]:
```suggestion
declare(strict_types=1);

```

File: src/Entity/ProductCommentCriterionCategory.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
```
Comment (Progi1984) [inline src/Entity/ProductCommentCriterionCategory.php:26]:
```suggestion
declare(strict_types=1);

```

File: src/Entity/ProductCommentCriterionLang.php [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
```
Comment (Progi1984) [inline src/Entity/ProductCommentCriterionLang.php:26]:
```suggestion
declare(strict_types=1);

```

File: src/Entity/ProductCommentCriterionProduct.php [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
```
Comment (Progi1984) [inline src/Entity/ProductCommentCriterionProduct.php:26]:
```suggestion
declare(strict_types=1);

```

**Comment by PierreRambaud** [comment]
Hi @jolelievre ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**